src/core.js

```diff
@@ -188,11 +188,11 @@ const patchOuter = patchFactory(function(node, fn, data) {
  * key.
  *
  * @param {?string} nodeName The nodeName for this node.
+ * @param {Element} matchNode A node to match the data to.
  * @param {?string=} key An optional key that identifies a node.
- * @param {Element=} matchNode An optional node to match the data to.
  * @return {boolean} True if the node matches, false otherwise.
  */
-const matches = function(nodeName, key, matchNode) {
+const matches = function(nodeName, matchNode, key) {
```

Comment:
Makes a bit more sense to move `matchNode` first to keep `nodeName` and `key` together.